tools/htseq.wdl

```diff
@@ -9,24 +9,52 @@ task count {
         String inferred_strandedness = ""
-        String outfile = basename(bam, ".bam") + ".feature-counts.txt"
-        Int added_memory_gb = 20
+        Boolean pos_sorted = true
+        Int minaqual = 10
+        String feature_type = "exon"
+        String idattr = "gene_name"
+        String mode = "union"
+        Boolean nonunique = false
+        Boolean secondary_alignments = false
+        Boolean supplementary_alignments = false
+        Int added_memory_gb = 0
         Int max_retries = 1
     }
 
+    parameter_meta {
+        bam: "Input BAM format file to generate coverage for"
+        gtf: "Input genomic features in GTF format to count reads for"
+        outfile_name: "Name for the feature count file"
+        provided_strandedness: "<Stranded-Reverse/Stranded-Forward/Unstranded> User provided strandedness."
+        inferred_strandedness: "<Stranded-Reverse/Stranded-Forward/Unstranded> `ngsderive` derived strandedness."
+        pos_sorted: "Is the BAM position sorted?"
+        minaqual: "Skip all reads with alignment quality lower than the given minimum value"
+        feature_type: "Feature type (3rd column in GFF file) to be used, all features of other type are ignored"
+        idattr: "GFF attribute to be used as feature ID"
+        mode: "<union/intersection-strict/intersection-nonempty> Mode to handle reads overlapping more than one feature"
+        nonunique: "Score reads that align to or are assigned to more than one feature?"
+        secondary_alignments: "Score secondary alignments (SAM flag 0x100)?"
+        supplementary_alignments: "Score supplementary/chimeric alignments (SAM flag 0x800)?"
+        added_memory_gb: "Additional RAM to allocate for task. Default RAM is allocated dynamically based on the BAM size."
+        max_retries: "Number of times to retry in case of failure"
+    }
+
     String stranded = if (provided_strandedness != "") then 
                         if (provided_strandedness == "Stranded-Reverse") then "reverse" else
                         if (provided_strandedness == "Stranded-Forward") then "yes" else
                         if (provided_strandedness == "Unstranded") then "no"
-                        else "unknown-strand"
+                         else "unknown-strand" # this will intentionally cause htseq to error.
+                                               # You will need to manually specify in this case
                       else 
                         if (inferred_strandedness == "Stranded-Reverse") then "reverse" else
                         if (inferred_strandedness == "Stranded-Forward") then "yes" else 
                         if (inferred_strandedness == "Unstranded") then "no" else
-                        "unknown-strand" # this will intentionally cause htseq to error. You will need to manually specify
-                                         # in this case
+                        if (inferred_strandedness == "Inconclusive") then "no"
```

Comment:
Also accepting suggestions for better terminology than "high" and "low" (which refers to where the sample falls on the figures I've made. Without that visual context, the terms aren't very informative).

Reply:
I'm not sure the best way to communicate these decisions to the user. We output the `strandedness` results, so theoretically a user should be able to figure out "o, my sample got called `unstranded`, it must've been run as `unstranded` in HTSeq". But maybe that's putting too much faith in users.

How do you think we can make these decisions more explicit to the user?

Reply:
@adthrasher maybe we should defer this decision to another PR after some additional investigation and discussion with a broader group. I think in the long term, we want to be running `Inconclusive` samples without failing them. The exact mode they should be run under probably requires some investigation of how the results will differ (I'd guess `unstranded` is appropriate, but that's not a very "educated" guess).

So my proposal is to revert this change for now. We go back to failing `Inconclusive` samples to keep consistent with rnaseq-standard v2.
This also partially solves the issue of informing the user which mode was run. Inconclusive runs will fail (though is it properly communicated why? I think we need a more explicit log message).

Do you think we should be more explicit about which mode was run for all cases where it's been left up to ngsderive? I'd consider putting the mode run into the file name. What do you think? I'm not immediately coming up with any other solutions. WDL doesn't provide many ways to communicate with users, and I think file names might be our best bet.

Reply:
I would agree with rolling back to the prior implementation. Do we return the value of strandedness from the pipeline? At least then it would be explicitly displayed in the JSON of outputs.

Reply:
Looking at the workflow, we just return the file. It might be good to return the string representation as well. It is redundant, but at least then a user is more likely to see it, rather than if it is just buried in a file.

Reply:
I implemented the suggested change (outputting strandedness as a string in addition to a file). Now I'm curious to see how `miniwdl` handles that. I don't think I've ever run something with a string as output before. Depending on `miniwdl`'s behavior, we might want to rollback this decision.